### PR TITLE
Hide sync options if not the owner #10967

### PR DESCRIPTION
**Pull request by ivanmalagon**
PR related to issue #10967 

While fixing this, we ran into another issue that will be fixed also in this PR. 

More info at https://github.com/CartoDB/cartodb/issues/10967#issuecomment-265495916

Wait to merge until that is fixed also.

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/helpers/required-opts.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+var _ = require('underscore');
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/helpers/required-opts.js]: cc @CartoDB/frontend

File: lib/assets/javascripts/cartodb3/editor/layers/layer-header-view.js [hunk 2]
```diff
@@ -121,11 +117,20 @@ module.exports = CoreView.extend({
...
+    }
+  },
+
+  _createSyncInfo: function (tableModel) {
+    var syncModel = tableModel.getSyncModel();
     this._syncInfoView = new SyncInfoView({
       modals: this._modals,
-      syncModel: this._tableNodeModel._syncModel,
-      tableModel: this._tableNodeModel
+      syncModel: tableModel._syncModel,
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-header-view.js]: ```syncModel: syncModel,``` ?

**Comment by xavijam** [comment]
Tests passed, merging! 💃 , great job @ivanmalagon.

File: lib/assets/javascripts/cartodb3/helpers/required-opts.js (hunk 1 above)
Comment (matallo) [inline lib/assets/javascripts/cartodb3/helpers/required-opts.js]: @xavijam what?

Comment (matallo) [inline lib/assets/javascripts/cartodb3/helpers/required-opts.js]: Oh! I see now, I thought the comment was regarding the line, and not the file itself, cool!

Comment (ivanmalagon) [inline lib/assets/javascripts/cartodb3/helpers/required-opts.js]: I didn't know how to address the whole team, thanks @xavijam for it.

I've added this module `required-opts.js` just to not copy the same code in every model.

For using it,

```
var REQUIRED_OPTS = [
    'userActions',
    'useModel'
]
  
initialize: function (opts) {
  checkAndBuildOpts(opts, REQUIRED_OPTS, this);

  ...
}
```

**Pull request merged**